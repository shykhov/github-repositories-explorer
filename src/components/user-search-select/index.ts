export * from './user-search-select';
