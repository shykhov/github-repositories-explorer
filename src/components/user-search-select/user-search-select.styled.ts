import styled from 'styled-components';
import ReactSelect from 'react-select';

export const Select = styled(ReactSelect)`
  font-family: 'Lato', sans-serif;
  & > div {
    min-height: 48px;
    font-size: 16px;
  }
`;

export const customSelectStyles = {
  control: (provided: any) => ({
    ...provided,
    minHeight: '48px',
    border: 'none',
    borderColor: 'transparent !important',
    boxShadow: 'none',
  }),
  placeholder: (provided: any) => ({
    ...provided,
    marginLeft: '45px',
    color: 'hsl(0,0%,71%)',
  }),
  input: (provided: any) => ({
    ...provided,
    marginLeft: '45px',
  }),
};
