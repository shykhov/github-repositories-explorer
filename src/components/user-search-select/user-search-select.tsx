import React from 'react';

import { Select, BarWrapper } from './user-search-select.styled';
import { SearchOption } from './search-option';

interface InputOptions {
  inputValue?: string;
}

export interface SelectValue {
  value: string;
  label: string;
  iconSrc: string;
}

interface SearchBarProps {
  onInputChange(): void;
  onSelectChange(user: SelectValue): void;
  value: SelectValue;
  loading: boolean;
  options: any;
}

export const UserSearchSelect = (props: SearchBarProps) => {
  const { onInputChange, onSelectChange, value, loading, options } = props;

  return (
    <BarWrapper>
      <Select
        isLoading={loading}
        value={value}
        cacheOptions
        isSearchable
        maxMenuHeight={200}
        components={{ Option: SearchOption, SingleValue: SearchOption }}
        isClearable
        onChange={onSelectChange}
        noOptionsMessage={({ inputValue }: InputOptions) =>
          !inputValue && !loading ? 'Type user name e.g. "Dan Abramov"' : null
        }
        placeholder="Search for github user name..."
        options={options}
        onInputChange={onInputChange}
      />
    </BarWrapper>
  );
};
