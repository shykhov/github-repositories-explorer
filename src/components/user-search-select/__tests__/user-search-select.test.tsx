import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { UserSearchSelect } from '../user-search-select';

describe('<UserSearchSelect />', () => {
  let renderedComponent: ShallowWrapper;
  let propsMock;
  const firstMockSelectValue = { value: 'test-value-1', label: 'test-label-1', iconSrc: 'test-icon-1' };
  const secondMockSelectValue = { value: 'test-value-2', label: 'test-label-2', iconSrc: 'test-icon-2' };

  beforeEach(() => {
    propsMock = {
      onInputChange: jest.fn(),
      onSelectChange: jest.fn(),
      value: firstMockSelectValue,
      loading: false,
      options: [firstMockSelectValue, secondMockSelectValue],
    };

    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    renderedComponent = shallow(<UserSearchSelect {...propsMock} />);
  });

  it('should render UserSearchSelect', () => {
    expect(renderedComponent.dive().dive()).toMatchSnapshot();
  });

  it('should render UserSearchSelect with true state', () => {
    renderedComponent.setProps({ loading: true });

    expect(renderedComponent.dive().dive()).toMatchSnapshot();
  });

  it('should render UserSearchSelect with empty value', () => {
    renderedComponent.setProps({ value: undefined });

    expect(renderedComponent.dive().dive()).toMatchSnapshot();
  });

  it('should render UserSearchSelect with empty options', () => {
    renderedComponent.setProps({ value: undefined });

    expect(renderedComponent.dive().dive()).toMatchSnapshot();
  });
});
