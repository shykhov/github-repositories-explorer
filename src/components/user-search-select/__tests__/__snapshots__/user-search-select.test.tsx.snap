// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserSearchSelect /> should render UserSearchSelect 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation10 MuiPaper-rounded"
>
  <Styled(StateManager)
    cacheOptions={true}
    components={
      Object {
        "Option": [Function],
        "SingleValue": [Function],
      }
    }
    isClearable={true}
    isLoading={false}
    isSearchable={true}
    maxMenuHeight={200}
    noOptionsMessage={[Function]}
    onChange={[MockFunction]}
    onInputChange={[MockFunction]}
    options={
      Array [
        Object {
          "iconSrc": "test-icon-1",
          "label": "test-label-1",
          "value": "test-value-1",
        },
        Object {
          "iconSrc": "test-icon-2",
          "label": "test-label-2",
          "value": "test-value-2",
        },
      ]
    }
    placeholder="Search for github user name"
    styles={
      Object {
        "control": [Function],
        "input": [Function],
        "placeholder": [Function],
      }
    }
    value={
      Object {
        "iconSrc": "test-icon-1",
        "label": "test-label-1",
        "value": "test-value-1",
      }
    }
  />
</div>
`;

exports[`<UserSearchSelect /> should render UserSearchSelect with empty options 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation10 MuiPaper-rounded"
>
  <Styled(StateManager)
    cacheOptions={true}
    components={
      Object {
        "Option": [Function],
        "SingleValue": [Function],
      }
    }
    isClearable={true}
    isLoading={false}
    isSearchable={true}
    maxMenuHeight={200}
    noOptionsMessage={[Function]}
    onChange={[MockFunction]}
    onInputChange={[MockFunction]}
    options={
      Array [
        Object {
          "iconSrc": "test-icon-1",
          "label": "test-label-1",
          "value": "test-value-1",
        },
        Object {
          "iconSrc": "test-icon-2",
          "label": "test-label-2",
          "value": "test-value-2",
        },
      ]
    }
    placeholder="Search for github user name"
    styles={
      Object {
        "control": [Function],
        "input": [Function],
        "placeholder": [Function],
      }
    }
  />
</div>
`;

exports[`<UserSearchSelect /> should render UserSearchSelect with empty value 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation10 MuiPaper-rounded"
>
  <Styled(StateManager)
    cacheOptions={true}
    components={
      Object {
        "Option": [Function],
        "SingleValue": [Function],
      }
    }
    isClearable={true}
    isLoading={false}
    isSearchable={true}
    maxMenuHeight={200}
    noOptionsMessage={[Function]}
    onChange={[MockFunction]}
    onInputChange={[MockFunction]}
    options={
      Array [
        Object {
          "iconSrc": "test-icon-1",
          "label": "test-label-1",
          "value": "test-value-1",
        },
        Object {
          "iconSrc": "test-icon-2",
          "label": "test-label-2",
          "value": "test-value-2",
        },
      ]
    }
    placeholder="Search for github user name"
    styles={
      Object {
        "control": [Function],
        "input": [Function],
        "placeholder": [Function],
      }
    }
  />
</div>
`;

exports[`<UserSearchSelect /> should render UserSearchSelect with true state 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation10 MuiPaper-rounded"
>
  <Styled(StateManager)
    cacheOptions={true}
    components={
      Object {
        "Option": [Function],
        "SingleValue": [Function],
      }
    }
    isClearable={true}
    isLoading={true}
    isSearchable={true}
    maxMenuHeight={200}
    noOptionsMessage={[Function]}
    onChange={[MockFunction]}
    onInputChange={[MockFunction]}
    options={
      Array [
        Object {
          "iconSrc": "test-icon-1",
          "label": "test-label-1",
          "value": "test-value-1",
        },
        Object {
          "iconSrc": "test-icon-2",
          "label": "test-label-2",
          "value": "test-value-2",
        },
      ]
    }
    placeholder="Search for github user name"
    styles={
      Object {
        "control": [Function],
        "input": [Function],
        "placeholder": [Function],
      }
    }
    value={
      Object {
        "iconSrc": "test-icon-1",
        "label": "test-label-1",
        "value": "test-value-1",
      }
    }
  />
</div>
`;
